timelock-server/src/integTest/java/com/palantir/atlasdb/timelock/AsyncTimeLockServiceMetadataIntegrationTest.java

```diff
@@ -0,0 +1,281 @@
+    private static final LockWatchEventVisitor LOCK_WATCH_EVENT_VISITOR = new LockWatchEventVisitor();
+    private static final LockWatchStateUpdateVisitor LOCK_WATCH_STATE_UPDATE_VISITOR =
+            new LockWatchStateUpdateVisitor();
+
+    private static final String WATCHED_TABLE_NAME = "watched-table";
+    private static final LockDescriptor WATCHED_LOCK_1 =
+            AtlasRowLockDescriptor.of(WATCHED_TABLE_NAME, "lock1".getBytes(StandardCharsets.UTF_8));
+    private static final LockDescriptor WATCHED_LOCK_2 =
+            AtlasRowLockDescriptor.of(WATCHED_TABLE_NAME, "lock2".getBytes(StandardCharsets.UTF_8));
+    private static final LockDescriptor WATCHED_LOCK_3 =
+            AtlasRowLockDescriptor.of(WATCHED_TABLE_NAME, "lock3".getBytes(StandardCharsets.UTF_8));
+    private static final LockDescriptor WATCHED_LOCK_4 =
+            AtlasRowLockDescriptor.of(WATCHED_TABLE_NAME, "lock4".getBytes(StandardCharsets.UTF_8));
+    private static final Map<LockDescriptor, ChangeMetadata> ALL_WATCHED_LOCKS_WITH_METADATA = ImmutableMap.of(
+            WATCHED_LOCK_1,
+            ChangeMetadata.unchanged(),
+            WATCHED_LOCK_2,
+            ChangeMetadata.updated("old".getBytes(StandardCharsets.UTF_8), "new".getBytes(StandardCharsets.UTF_8)),
+            WATCHED_LOCK_3,
+            ChangeMetadata.deleted("deleted".getBytes(StandardCharsets.UTF_8)),
+            WATCHED_LOCK_4,
+            ChangeMetadata.created("created".getBytes(StandardCharsets.UTF_8)));
+    private static final String UNWATCHED_TABLE_NAME = "a-random-unwatched-table";
+    private static final LockDescriptor UNWATCHED_LOCK_1 =
+            AtlasRowLockDescriptor.of(UNWATCHED_TABLE_NAME, "lock1".getBytes(StandardCharsets.UTF_8));
+    private static final IdentifiedLockRequest WATCHED_LOCK_REQUEST_WITH_METADATA =
+            standardRequestWithMetadata(ALL_WATCHED_LOCKS_WITH_METADATA);
+
+    private final LockLog lockLog = new LockLog(new MetricRegistry(), () -> 10000L);
+    private final ScheduledExecutorService scheduledExecutorService = PTExecutors.newSingleThreadScheduledExecutor();
+    private final AsyncLockService asyncLockService =
+            AsyncLockService.createDefault(lockLog, scheduledExecutorService, scheduledExecutorService);
+    private final AsyncTimelockServiceImpl timeLockService =
+            new AsyncTimelockServiceImpl(asyncLockService, new InMemoryTimestampService(), lockLog);
+    private final ConjureStartTransactionsRequest startTransactionsRequestWithInitialVersion =
+            ConjureStartTransactionsRequest.builder()
+                    .requestId(UUID.randomUUID())
+                    .lastKnownVersion(ConjureIdentifiedVersion.of(
+                            asyncLockService.leaderTime().id().id(), 0))
+                    .numTransactions(1)
+                    .requestorId(UUID.randomUUID())
+                    .build();
+
+    @Before
+    public void setupLockServiceWithWatches() {
```

Comment:
my bias is to just call this `setup`/`before`, although I see you probably gleaned this from other tests. reason why my bias is towards a simple naming, is that in case in the future you need to setup additional methods/classes, you can add them here; you probably don't want multiple `@before` as the ordering is arbitrary. 

that said, I don't feel strongly about it :)